### DISCUSSION TITLE
core clock limit value change code?

### DISCUSSION THREAD

**Discussion by hadixcanimm** [seerge/g-helper#2500] [General]
Is there any code I can write in the config.json file to change the maximum and minimum value of core clock limit or can you add it?

![Ekran görüntüsü 2024-04-26 161505](https://github.com/seerge/g-helper/assets/114208757/df7b6b20-1eee-4f96-beb9-64f2f4c35902)

**Comment by seerge** [comment]
@hadixcanimm hello,

Current clock limit range is 
```
    public const int MinClockLimit = 400;
    public const int MaxClockLimit = 3000;
```

Why would you need something outside of that range?

**Reply by hadixcanimm** [reply]
My device doesn't go higher than 2100MHz, so I want to set the maximum limit to this value. I believe by locking it to a minimum of 300MHz, I can achieve a quieter gaming experience. Low-power games like LoL and Valorant even allow me to reach 144 FPS at 400MHz. Frankly, I'm also curious about the lower limit value.

Where do I need to write this code you provided?

Into the AppData\Roaming\Ghalper config.json, right?

**Reply by seerge** [reply]
@hadixcanimm there is no "lower limit" :) and you can't set that even ...

What i linked to you is a range for a ``Core Clock Limit`` slider - i.e. you can set **Maximum** value within that range.

Just set it to 400Mhz if you really want to restrict your device.

You don't need to write anything into config for that.

**Reply by hadixcanimm** [reply]
Can I adjust this setting with a code like 'MinClockLimit', between 300 and 400 there isn't much difference, but I'm curious about how our device will react.

I've set our device's 'clock limit' value to 1350 MHz and entered Valorant. I can set the memory frequency value to at least -1000, but Galper doesn't allow -1001.

I've tested with the following values in Valorant:

- Core Clock: 1350 MHz
- Memory Clock: 5000 MHz
- GPU Power Consumption: 16 watts
- GPU Temperature: 60 degrees

However, if I set the limitations as follows:

- Core Clock: 400
- Memory Clock: 5000

When I set the core clock limit to minimum, my FPS remains at 144 FPS. But I think sometimes the device thinks it's in low-power mode and reduces the memory clock limit to 800 MHz, even though it's set to the limit. My FPS still remains at 144 FPS, but here's the interesting part:

- Core Clock: 400
- Memory Clock: 800
- GPU Power Consumption: 5 watts
- GPU Temperature: 45 degrees

So, the device runs very cool and quietly and consumes almost three times less power. Probably due to setting the memory clock limit to 800, which I can't change manually, but the device does it when it thinks it's consuming less power.

Maybe if I can set the core clock limit to 300 or lower, the memory clock can run continuously at 800 MHz. Therefore, I want to try a lower setting. I wish there was a way.

**Reply by seerge** [reply]
@hadixcanimm does that limit even work for you ? Can you run afterburner or some kind of overlay in game and check if actual Core Clock of your GPU is indeed < 400Mhz ?

I can allow lower slider values, if needed. I just want to understand if you are looking at a right thing :)

**Reply by hadixcanimm** [reply]
Yes, it's working. I think the minimum value of my device is 300 MHz, but I'm curious if I can set it lower.

**Reply by hadixcanimm** [reply]
![lowpower](https://github.com/seerge/g-helper/assets/114208757/963c6a29-cdbb-4225-afc7-fbfdf6e203cc)
![normal](https://github.com/seerge/g-helper/assets/114208757/56fb3d66-1666-4687-933e-a1ea251fab0b)

**Reply by hadixcanimm** [reply]
the temperatures are the same because there are only a few seconds in between.

**Reply by seerge** [reply]
@hadixcanimm try this build

[GHelper.zip](https://github.com/seerge/g-helper/files/15133826/GHelper.zip)

**Reply by hadixcanimm** [reply]
yes build worked. 

can you please adapt this to the build where you do manual fan mode? i was using manual fan build it works great for me.
Also
Is there any code I can tweak on config.json?

For example:
"MinClockLimit": 200;

Like I said, I'm curious about the limits of the gpu.

**Reply by seerge** [reply]
@hadixcanimm I have updated https://github.com/seerge/g-helper/discussions/2272 with latest changes

And also added optional param for config called
```
"min_gpu_clock" : 400,
```

You can add and set even to 0
